src/event.js

```diff
@@ -91,6 +91,36 @@ function on( elem, types, selector, data, fn, one ) {
 	} );
 }
 
+function addEventGetter( unused, name ) {
+	Object.defineProperty( jQuery.Event.prototype, name, {
+		enumerable: true,
+		configurable: true,
+
+		get: function() {
+			var value, hook;
+
+			if ( this.originalEvent ) {
+				if ( hook = jQuery.event.propHooks[ name ] ) {
+					value = hook( this.originalEvent );
+				} else {
+					value = this.originalEvent[ name ];
+				}
+			}
+
+			return this[ name ] = value;
+		},
+
+		set: function( value ) {
+			Object.defineProperty( this, name, {
```

Comment:
Rather than defining a setter on property access which can be a slow operation, would it be better to store the final value under a separate key on `this` and just return that in the getter if it's been defined?